addon/mixins/route.js

```diff
@@ -199,6 +200,17 @@ export default Ember.Mixin.create({
   },
 
   /**
+   Call additional functions after finding the infinityModel in the Ember data store.
+
+   @method afterInfinityModel
+   @param {Function} infinityModelPromise The resolved result of the Ember store find method. Passed in automatically.
+   @return {Ember.RSVP.Promise}
+   */
+  afterInfinityModel(infinityModelPromiseResult) {
+    return infinityModelPromiseResult;
```

Comment:
aah not exactly. There's already an `infinityModelUpdated` hook that executes async, and it doesn't work for #90 because the user route needs to update itself synchronously, before another request can be made.

If we go with this PR, I'd propose treating `afterInfinityModel` EXACTLY the same as `Ember.Route#afterModel` (http://emberjs.com/api/classes/Ember.Route.html#method_afterModel), if the value returned from this hook is a promise, the transition will pause until the transition resolves. Otherwise, non-promise return values are not utilized in any way.